DQM/BeamMonitor/plugins/BeamSpotProblemMonitor.cc

```diff
@@ -189,182 +150,156 @@ void BeamSpotProblemMonitor::analyze(const Event& iEvent,
+  }
 
-   if(BeamSpotFromDB_ >= alarmONThreshold_ ){ ALARM_ON_ =true; //set the audio alarm true after N successive LSs
-                                           }
+  if(beamSpotFromDB_ > alarmOFFThreshold_ ){
+    alarmOn_ =false; //set the audio alarm true after 10 successive LSs
+    beamSpotFromDB_=0; //reset it for new incident
+  }
 
-   if(BeamSpotFromDB_ > alarmOFFThreshold_ ){ ALARM_ON_ =false; //set the audio alarm true after 10 successive LSs
-                                               BeamSpotFromDB_=0; //reset it for new incident
-                                             }
 
 
+  if (onlineMode_)  { // filling LS gap For status plot
 
- if (onlineMode_) 
- { // filling LS gap For status plot
-    const int countLS_bs = hs["BeamSpotStatus_lumi"]->getTH1()->GetEntries();
+    const int countLS_bs = beamSpotStatusLumi_->getTH1()->GetEntries();
     int LSgap_bs = currentlumi/fitNLumi_ - countLS_bs;
     if (currentlumi%fitNLumi_ == 0)LSgap_bs--;
 
 
     // filling previous fits if LS gap ever exists
     for (int ig = 0; ig < LSgap_bs; ig++) {
-      hs["BeamSpotStatus_lumi"]->ShiftFillLast( 0., 0., fitNLumi_ );//x0 , x0err, fitNLumi_;  see DQMCore....
-     }
-
-     hs["BeamSpotStatus_lumi"]->ShiftFillLast( BeamSpotStatus_, 0. , fitNLumi_ ); //BeamSpotStatus_ =>0. (no collision, no tracks); =>1 (OK from scaler), =>-1 (No scalar results) 
-     hs["BeamSpotStatus_lumi_all"]->setBinContent( currentlumi, BeamSpotStatus_);
-
-
- }//onlineMode_
- else { 
-      hs["BeamSpotStatus_lumi"]->ShiftFillLast( 0., 0., fitNLumi_ );
-     }
+      beamSpotStatusLumi_->ShiftFillLast( 0., 0., fitNLumi_ );//x0 , x0err, fitNLumi_;  see DQMCore....
+    }
 
-      //Reset it here for next lumi
-      BeamSpotError->Reset();
-      if(ALARM_ON_)BeamSpotError->Fill(BeamSpotFromDB_);
+    beamSpotStatusLumi_->ShiftFillLast( beamSpotStatus_, 0. , fitNLumi_ ); //beamSpotStatus_ =>0. (no collision, no tracks); =>1 (OK from scaler), =>-1 (No scalar results)
+    beamSpotStatusLumiAll_->setBinContent( currentlumi, beamSpotStatus_);
 
 
-    //Get quality report
-     MonitorElement* myQReport = dbe_->get(monitorName_+"FitFromScalars/BeamSpotError");
+  } else {
+    beamSpotStatusLumi_->ShiftFillLast( 0., 0., fitNLumi_ );
+  }//onlineMode_
 
-     const QReport * BeamSpotQReport = myQReport->getQReport("BeamSpotOnlineTest");  
+   //Reset it here for next lumi
+  beamSpotError_->Reset();
+  if(alarmOn_)beamSpotError_->Fill(beamSpotFromDB_);
 
-    if(BeamSpotQReport){  
-    		          /* S.Dutta : Commenting out these variables are not used and giving error with "-Werror=unused-variable" option
-    	                   float qtresult = BeamSpotQReport->getQTresult();
-                           int qtstatus   = BeamSpotQReport->getStatus() ; // get QT status value (see table below) */
-                      }
 
+  //Get quality report
+  const QReport * beamSpotQReport = beamSpotError_->getQReport("BeamSpotOnlineTest");
```

Comment:
Does this actually do anything useful? I was concerned this code is intended to cause a side-effect which is why I did not remove it.